app/controllers/users.py

```diff
@@ -376,20 +377,38 @@ def delete(self, user_id):
         except Exception as e:
             return dict(status='fail', message=str(e)), 500
 
+    @jwt_required
     def patch(self, user_id):
         """
         """
         try:
-            user_schema = UserSchema(only=("name",))
+            user_schema = UserSchema()
 
             user_data = request.get_json()
 
-            validate_user_data, errors = user_schema.load(user_data)
+            current_user_id = get_jwt_identity()
+            current_user_roles = get_jwt_claims()['roles']
+            
+            if (current_user_id != user_id):
+                if (not is_admin(current_user_roles)):
+                    return dict(
+                        status = 'UnAuthorised',
+                        message = 'You are not authorized to edit this users information'
+                    ) , 401
+
+            user = User.get_by_id(user_id)
+
+            if (not user) :
+                return dict(
+                    status = 'Not found',
+                    message = f'User not with id {user_id} not found'
+                ),404
+
+            validate_user_data, errors = user_schema.load({**user.__dict__,**dict(user_data)})
```

Comment:
I think you only need to load the schema with data that has been sent from the request object 
```suggestion
            validate_user_data, errors = user_schema.load(user_data)
```